### PR TITLE
Remove useless ShutdownHook

### PR DESCRIPTION
**Pull request by cheese8**
Ref https://github.com/apache/shardingsphere/issues/10641

Changes proposed in this pull request:

-try to enhance it, please figure it out any problems

### CODE REVIEW

**Comment by terrymanu** [comment]
What is the usage to remove `SHUTDOWN_EXECUTOR`?

**Comment by cheese8** [comment]
> What is the usage to remove `SHUTDOWN_EXECUTOR`?

The original usage is register a delayed shutdown hook, while closing `ExecutorEngine` which cause by `AutoCloseable` using another thread `SHUTDOWN_EXECUTOR` to shutdown the `executorService` asynchronously.

I think it should be break by while closing `ExecutorEngine` and `UserExecutorGroup` which cause by `AutoCloseable` shutdown the `executorService` directly.

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/UserExecutorGroup.java [hunk 1]
```diff
@@ -46,4 +48,21 @@ private UserExecutorGroup() {
...
+        if (null != executorService) {
+            try {
+                executorService.shutdown();
+                while (!executorService.awaitTermination(5, TimeUnit.SECONDS)) {
+                    executorService.shutdownNow();
+                }
+            } catch (final InterruptedException ex) {
+                Thread.currentThread().interrupt();
+            }
+        }
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/UserExecutorGroup.java:66]: UserExecutorGroup is for proxy only, spring boot maybe not need change

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/UserExecutorGroup.java (hunk 1 above)
Comment (cheese8) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/UserExecutorGroup.java]: ok, revert it.

**Review by terrymanu — APPROVED** [review]

**Comment by terrymanu** [comment]
We are talking about in PR

**Pull request merged**